src/opaque.rs

```diff
@@ -104,37 +112,39 @@ where
     type Error = ProtocolError;
 
     fn try_from(second_message_bytes: &[u8]) -> Result<Self, Self::Error> {
-        let checked_slice = check_slice_size(
-            second_message_bytes,
-            Grp::ElemLen::to_usize(),
-            "second_message_bytes",
-        )?;
-        // Check that the message is actually containing an element of the
-        // correct subgroup
-        let arr = GenericArray::from_slice(&checked_slice);
-        let beta = Grp::from_element_slice(arr)?;
-        Ok(Self { beta })
+        Self::deserialize(second_message_bytes)
     }
 }
 
 impl<Grp> RegisterSecondMessage<Grp>
 where
     Grp: Group,
 {
-    /// byte representation for the registration response message
-    fn to_bytes(&self) -> Vec<u8> {
-        self.beta.to_arr().to_vec()
-    }
-
```

Comment:
lost method

Reply:
Same as above (I think we should simply disallow a way to get bytes without calling serialize())